ClinGen/clincoded
# Connects to #482, Connects to #328, Connects to #511, Connects to #512, Connects to #504

**Changes for Ticket #328 and Ticket #511**
- Edit titles at data entrance boxes of Orphanet Disease, HPO id and Phenotype free text in group and family pages.
- Add a description for HPO and free text in group, family, individual and experimental pages.
![summary](https://cloud.githubusercontent.com/assets/9206696/12212102/4f82e620-b61b-11e5-98cc-2e15560fd399.png)

**Change for Ticket #512**
- Edit title of method field in group, family and individual pages.
![summary](https://cloud.githubusercontent.com/assets/9206696/12209825/83f44fd6-b60a-11e5-8f87-c89600004d9a.png)


**Changes for Ticket #482 and Ticket #504:**
- Remove property commonDiagnosis from required list in group schema.
- Add controls for error handling when creating or editing a group.
- Add controls allowing delete Orphanet, hpo and free text in group, family and individual.
- Add controls for displaying and copying group Orphanet and phenotype to family and individual.
- Add controls for displaying and copying family Orphanet and phenotype to individual.
- Use empty string for search in case of no Orphanet entered in family.js and individual.js. No interface change (Ticket #504).

**Test Steps for #482 and #504**
* Create GDM.
* Add a PMID to the GDM.
* Add a group to the PMID, name it Group100, enter a number at field "Total number individuals in group", leave Orphanet, HPO and Free Text fields empty.
* Click Save and expect to see error messages:
![summary](https://cloud.githubusercontent.com/assets/9206696/12209714/cb1918ac-b609-11e5-8b45-0675b503330d.png)
* Enter ORPHA891 and expect to see all error messages disappear.
* Save the group.
* Add new family to the group, name it Family110. Expect to see group Orphanet and a copy button, don't click it.
![summary](https://cloud.githubusercontent.com/assets/9206696/12210331/01f61bfa-b60e-11e5-8c91-ab08b5502d3a.png)
* Add variant 111 to the family and a proband individual Ind111. Expect to see no family Orphanet and no copy button in Family-Variant panel.
![summary](https://cloud.githubusercontent.com/assets/9206696/12210475/2057fa9a-b60f-11e5-9da0-8b4a732551ed.png)
* Click button "Copy Orphanet IDs from Associated Group", expect to see Orphnet id copied to family.
![summary](https://cloud.githubusercontent.com/assets/9206696/12210684/a54bfa48-b610-11e5-96a8-551ddff90f3d.png)
In addition, family Orphanet and a copy button are displayed for the proband individual in Family - Variant panel.
![summary](https://cloud.githubusercontent.com/assets/9206696/12210655/6b604848-b610-11e5-867b-ae33a7d51469.png)
* Copy the Orphnet to the individual.
* Save the family and go back to curation-central page.
* Add new individual to the group, expect to see group Orphanet id and the copy button.
![summary](https://cloud.githubusercontent.com/assets/9206696/12211396/82659cdc-b615-11e5-9d47-d9bb8235eb92.png)
* Name the individual as Ind101, select non-proband, copy group Orphnet and save it.
* Edit Group100, enter a HPO id and save.
* Edit Family110. Expect to see group HPO id and button "Copy Phenotype from Associated Family".
![summary](https://cloud.githubusercontent.com/assets/9206696/12210887/1df7e154-b612-11e5-95d0-5ef8a4262c01.png)
* Click the button and expect to see the HPO id copied to the family.
* Click Cancel.
* Edit Ind101. Expect to see the same HPO id and button in individual edit page.
* Click the button and expect to see the HPO id copied to the individual.
* Click Cancel.
* Edit Group100, enter a phenotype free text and save. 
* Edit Family110, expect to see group HPO id followed by "View Group100 for phenotype free text."
* Click link Group100, expect to open a view page of the group.
* Click "Copy Phenotype from Associated Group" button, expect to see the text copied to the family.
* Click Cancel.
* Edit Ind101, expect to see the same HPO id and message as above.
* Click the copy button and expect to see group free text copied to Ind101.
* Click Cancel.
* Edit Group100, change its Orphanet ID, HPO ID and free text.
* Edit Family110, expect to see the new group Orphanet ID and HPO ID is shown.
* Click copy buttons, expect to see group terms copied to the family. Cancel.
* Edit Ind101, expect to see the same group terms in individual page.
* Click copy buttons, expect to see group terms copied to the individual. Cancel.
* Edit Group100, delete Orphanet ID and click Save.
* Edit Family110, expect not to see group Orphanet and button "Copy Orphanet IDs from Associated Group".
* Manually delete Orphanet id in the family, click Save, #504(search string) test here.
* Edit Ind101, expect same result as above.
* Manually delete Orphanet id click Save, #504 test here.
* Add new individual to Family110. Expect not to see Orphanet Id, HPO id, phenotype free text and copy buttons, click Cancel.
* Edit Family110, enter Orphanet id, HPO id and phenotype free text and save.
* Add new individual to Family110, expect to see family Orphanet, HPO, free text and 2 copy buttons.

**End Test**
